dot/parachain/types/subsystems.go

```diff
@@ -0,0 +1,552 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package parachaintypes
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+type SubSystemName string
+
+const (
+	CandidateBacking  SubSystemName = "CandidateBacking"
+	CollationProtocol SubSystemName = "CollationProtocol"
+	AvailabilityStore SubSystemName = "AvailabilityStore"
+)
+
+var (
+	_ ProvisionerMessage           = (*PMProvisionableData)(nil)
+	_ ProvisionableData            = (*PDBackedCandidate)(nil)
+	_ ProvisionableData            = (*PDMisbehaviorReport)(nil)
+	_ StatementDistributionMessage = (*SDMBacked)(nil)
+	_ CollatorProtocolMessage      = (*CPMBacked)(nil)
+	_ ProspectiveParachainsMessage = (*PPMCandidateBacked)(nil)
+	_ ProspectiveParachainsMessage = (*PPMIntroduceCandidate)(nil)
+	_ ProspectiveParachainsMessage = (*PPMCandidateSeconded)(nil)
+	_ RuntimeApiMessage            = (*RAMRequest)(nil)
+	_ RuntimeApiRequest            = (*RARValidationCodeByHash)(nil)
+	_ CandidateValidationMessage   = (*CVMValidateFromExhaustive)(nil)
+	_ AvailabilityStoreMessage     = (*ASMStoreAvailableData)(nil)
+	_ Misbehaviour                 = (*IssuedAndValidity)(nil)
+	_ ValidityDoubleVote           = (*IssuedAndValidity)(nil)
+	_ Misbehaviour                 = (*MultipleCandidates)(nil)
+	_ Misbehaviour                 = (*UnauthorizedStatement)(nil)
+	_ Misbehaviour                 = (*OnSeconded)(nil)
+	_ DoubleSign                   = (*OnSeconded)(nil)
+	_ Misbehaviour                 = (*OnValidity)(nil)
+	_ DoubleSign                   = (*OnValidity)(nil)
+)
+
+// OverseerFuncRes is a result of an overseer function
+type OverseerFuncRes[T any] struct {
+	Err  error
+	Data T
+}
+
+// ProvisionerMessage is a message to the Provisioner.
+type ProvisionerMessage interface {
+	IsProvisionerMessage()
+}
+
+// PMProvisionableData is a provisioner message.
+// This data should become part of a relay chain block.
+type PMProvisionableData struct {
+	RelayParent       common.Hash
+	ProvisionableData ProvisionableData
+}
+
+func (PMProvisionableData) IsProvisionerMessage() {}
+
+// ProvisionableData becomes intrinsics or extrinsics which should be included in a future relay chain block.
+type ProvisionableData interface {
+	IsProvisionableData()
+}
+
+// PDBackedCandidate is a provisionable data.
+// The Candidate Backing subsystem believes that this candidate is valid, pending availability.
+type PDBackedCandidate CandidateReceipt
```

Comment:
let's create a file called `overseer_message.go` and move all overseer message types there.
That way if or when we want to move all overseer messages to `dot/parachain/overseer` we can do that easily.

So SDMBacked will become StatementDistributionMessageBacked in that file.